app/frontend/src/components/designer/FormDesigner.vue

```diff
@@ -334,6 +399,15 @@ export default {
           this.formSchema = JSON.parse(fileReader.result);
           // Key-changing to force a re-render of the formio component when we want to load a new schema after the page is already in
           this.reRenderFormIo += 1;
+          this.addPatchToHistory();
+          this.patchHistoryUndo = false;
+          this.patchHistoryRedo = false;
+          this.componentAdded = false;
+          this.componentRemoved = false;
+          this.componentUpdated = false;
+          this.componentMoved = false;
```

Comment:
You already have a `resetHistoryFlags` method defined. You should be using reusing that.

```js
this.resetHistoryFlags();
```